src/Apps/Collect/Components/Collection/Header/__tests__/Header.test.ts

```diff
@@ -0,0 +1,97 @@
+import { featuredArtistsEntityCollection, getFeaturedArtists } from "../index"
+
+const mockedFeaturedArtistsEntityCollection = featuredArtistsEntityCollection as jest.Mock
+
+describe("getFeaturedArtists", () => {
+  const mockMediator = jest.fn()
+
+  const collection = {
+    title: "KAWS: Toys",
+    query: {
+      gene_id: null,
+      artist_id: null,
+      artist_ids: ["4e934002e340fa0001005336"],
+    },
+    artworks: {
+      merchandisable_artists: [
+        {
+          id: "kaws",
+          _id: "4e934002e340fa0001005336",
+          name: "KAWS",
+          imageUrl:
+            "https://d32dm0rphc51dk.cloudfront.net/WhacjFyMKlMkNVzncPjlRA/square.jpg",
+          birthday: "1974",
+          nationality: "American",
+        },
+        {
+          id: "robert-lazzarini",
+          _id: "4f5f64c23b555230ac0003ae",
+          name: "Robert Lazzarini",
+          imageUrl:
+            "https://d32dm0rphc51dk.cloudfront.net/1npk1i_Xua5q8Hv0YOq_3g/square.jpg",
+          birthday: "1965",
+          nationality: "American",
+        },
+        {
+          id: "medicom",
+          _id: "58fe85ee275b2450a0fd2b51",
+          name: "Medicom",
+          imageUrl:
+            "https://d32dm0rphc51dk.cloudfront.net/jUMOidRmCQ0RyynXM_sFzQ/square.jpg",
+          birthday: "",
+          nationality: "",
+        },
+        {
+          id: "medicom-toy-slash-china",
+          _id: "5b9821af86c8aa21d364dde5",
+          name: "Medicom Toy/China",
+          imageUrl:
+            "https://d32dm0rphc51dk.cloudfront.net/npEmyaOeaPzkfEHX5VsmQg/square.jpg",
+          birthday: "",
+          nationality: "",
+        },
+      ],
+    },
+  }
+
+  it("returns the queried artists when there is explicit artist_ids", () => {
+    const results = getFeaturedArtists(
+      9,
+      collection,
+      true,
+      collection.artworks.merchandisable_artists,
+      mockMediator,
+      {}
+    )
+
+    expect(results.length).toEqual(1)
+  })
+
+  it("passes correct arguments featuredArtistsEntityCollection", () => {
+    getFeaturedArtists(
+      9,
+      collection,
+      true,
+      collection.artworks.merchandisable_artists,
+      mockMediator,
+      {}
+    )
+
+    expect(mockedFeaturedArtistsEntityCollection).toBeCalledWith(
```

Comment:
@l2succes I mock `featuredArtistsEntityCollection` function on line 3 but it looks like it's still using the actual function. Is there something obvious I'm missing?

Reply:
@xtina-starr Here's what you're looking for https://github.com/facebook/jest/issues/936#issuecomment-214556122

You're not actually mocking the function on line 3, you're only type casting it with Typescript. Here's what you actually want

```typescript
import all, { getFeaturedArtists } from "../index"
const mockedFeaturedArtistsEntityCollection = all.featuredArtistsEntityCollection = jest.fn()
```

Reply:
@l2succes ah that makes sense. The file I'm testing doesn't have a default export so could I change the above to something like this:

```TypeScript
const Header = require("../index")
jest.unmock("../index")
Header.featuredArtistsEntityCollection = jest.fn()
```